packages/emotion/src/withStyle.tsx

```diff
@@ -126,9 +140,35 @@ const withStyle = decorator(
             )
           : {}
       )
-      const makeStyleHandler = (...extraArgs: any[]) => {
+      /**
+       * This method generates and sets the styles for the component (it calls
+       * the `generateStyle` method in the decorator's parameter).
+       * @param generateStyleArgs An object that will be passed to the
+       *     `generateStyle` method.
+       * @param updateConfig An object that you can use to prevent unnecessary
+       *     style recalculations for performance reasons. If not supplied
+       *     everything will be recalculated.
+       */
+      const makeStyleHandler: MakeStyles = (
+        generateStyleArgs,
+        updateConfig
+      ) => {
+        if (updateConfig) {
+          let changed = false
+          for (const propToCheck in updateConfig.styledProps) {
```

Comment:
If we keep the 1 signature `makeStyles` like I mentioned above, then it can be used here like this:
```js
 const makeStyleHandler: MakeStyles = (config) => {
        const { updateConfig, ...generateStyleArgs } = config
...
```